Source/DOM classes/SVG-DOM/SVGHelperUtilities.m

```diff
@@ -490,15 +490,17 @@ +(CALayer *) newCALayerForPathBasedSVGElement:(SVGElement<SVGTransformable>*) sv
 		
 		/** Replace the return layer with a special layer using the URL fill */
 		/** fetch the fill layer by URL using the DOM */
-		SVGGradientLayer *gradientLayer = [self getGradientLayerWithId:fillId forElement:svgElement withRect:fillLayer.frame];
-
+		SVGGradientLayer *gradientLayer = [self getGradientLayerWithId:fillId forElement:svgElement withRect:fillLayer.frame
+										   transform:transformAbsolute];
+		
 		CAShapeLayer* maskLayer = [CAShapeLayer layer];
 		maskLayer.frame = localRect;
 		maskLayer.path = fillLayer.path;
 		maskLayer.fillColor = [UIColor blackColor].CGColor;
 		maskLayer.strokeColor = nil;
 		gradientLayer.mask = maskLayer;
-		
+		if ( [gradientLayer.type isEqualToString:kExt_CAGradientLayerRadial])
+			gradientLayer.maskPath = pathToPlaceInLayer;
```

Comment:
This should have been in my last check in, since i haven't worked on radial gradients.